Formula/souffle.rb

```diff
@@ -34,9 +37,27 @@ def install
     ]
     system "cmake", "-S", ".", "-B", "build", *cmake_args, *std_cmake_args
     inreplace "#{buildpath}/build/src/souffle-compile.py" do |s|
-      s.gsub!(/"compiler": ".*?"/, "\"compiler\": \"/usr/bin/c++\"")
-      s.gsub!(%r{-I.*?/src/include }, "")
-      s.gsub!(%r{"source_include_dir": ".*?/src/include"}, "\"source_include_dir\": \"#{include}\"")
+      sqlite_include = "#{Formula["sqlite"].opt_include}"
+      lib_ext = ".so"
+      lib_ext = ".dylib" if OS.mac?
+      link_options = [
+        "#{Formula["sqlite"].opt_lib}/libsqlite3#{lib_ext}",
+        "#{Formula["zlib"].opt_lib}/libz#{lib_ext}",
+        "#{Formula["ncurses"].opt_lib}/libncurses#{lib_ext}",
+      ]
+      if OS.linux?
+        link_options << "-ldl"
+      end
+      rpaths = [
+        "#{Formula["sqlite"].opt_lib}",
+        "#{Formula["zlib"].opt_lib}",
+        "#{Formula["ncurses"].opt_lib}",
+      ]
+      s.gsub!("#{Formula["sqlite"].opt_lib}", "#{Formula["sqlite3"].opt_lib}")
+      s.gsub!(/"includes": ".*?"/, "\"includes\": \"-I#{include} -I#{sqlite_include}\"")
+      s.gsub!(/"link_options": ".*?"/, "\"link_options\": \"#{link_options * " "}\"")
+      s.gsub!(/"rpaths": ".*?"/, "\"rpaths\": \"#{rpaths * ":"}\"")
+      s.gsub!(/"source_include_dir": ".*?"/, "\"source_include_dir\": \"#{include}\"")
```

Comment:
This should be done by CMake, not by a homebrew patch.

Reply:
`cmake` generates `souffle-compile.py` at build time. 
It contains absolute paths to the libraries depended on by souffle at runtime. 
During bottling, homebrew needs to put a placeholder in the paths to dependencies, so that when a user installs homebrew in a non-standard location, the dependencies can still have the correct path. 

The patch here is just replace the absolute paths to various dependencies with the path generated by homebrew `Formula["xxx"]`. 
The patch does not add or delete any dependencies declared in the `souffle-compile.py`.

Reply:
> It contains absolute paths to the libraries depended on by souffle at runtime.

This will be a problem in most package managers, if not all. This is the real problem that needs fixing here.